core/src/main/java/org/testcontainers/containers/VncRecordingContainer.java

```diff
@@ -87,19 +88,27 @@ protected void configure() {
         );
     }
 
+    @SneakyThrows
+    public void saveRecordingToFile(File file) {
+        try(InputStream inputStream = streamRecording()) {
+            Files.copy(inputStream, file.toPath(), StandardCopyOption.REPLACE_EXISTING);
+        }
+    }
+
     @SneakyThrows
     public InputStream streamRecording() {
+        fixRecordDuration();
+
         TarArchiveInputStream archiveInputStream = new TarArchiveInputStream(
                 dockerClient.copyArchiveFromContainerCmd(getContainerId(), RECORDING_FILE_NAME).exec()
         );
         archiveInputStream.getNextEntry();
         return archiveInputStream;
     }
 
-    @SneakyThrows
-    public void saveRecordingToFile(File file) {
-        try(InputStream inputStream = streamRecording()) {
-            Files.copy(inputStream, file.toPath(), StandardCopyOption.REPLACE_EXISTING);
-        }
+    private void fixRecordDuration() throws IOException, InterruptedException {
```

Comment:
Should we rename this method to `reencodeRecording`?

Reply:
It's done.